### ISSUE TITLE
Potential use of pytest parametrize in `tests_datasets.py`

### ISSUE THREAD

**Issue by cr458** [geopandas/geopandas#644]
I noticed some cases where `@pytest.mark.parametrize` would come in handy and clean up the testing code a little such as in `test_datasets.py`. Would be happy to make these changes if needed?

**Comment by jorisvandenbossche** [comment]
Yes, sure! We only started to use pytest recently for the new tests, so there are probably still quite some placed where it could be used. 
`test_datasets.py` is indeed such an example (and there it would also not be needed to use the class, just a single parameterized function would be good). PR welcome!

**Comment by cr458** [comment]
Great, will fix `test_datasets.py` first, and then continue through the tests in order to keep the PRs a bit more atomic. Unless there's a preference to parametrize all the tests that need it in one go?

**Comment by jorisvandenbossche** [comment]
Starting with that one is fine

**Comment by jdmcbr** [comment]
Potentially a separate issue of its own, but: thoughts on creating a conftest.py with test fixtures and using those in testing? It can obscure things a bit, because the definition of a test fixture lives in a separate file. That said, it would cut down on duplication in current setup methods for all the tests within each module. Moreover, variable names in general don't make clear what the setup variables are, so I, at least, go back and forth a bit between the setup definitions and the new test, where it might be easier to have the conftest.py open in one window and the new tests in another.

**Comment by cr458** [comment]
@jdmcbr I think that's a good idea. I find it can clean up the tests a fair amount, even though as you mentioned people unfamiliar with pytest may find it a little confusing ( I do find it a little magical how the fixtures are passed from the conftest.py file into the test functions).
Am happy to open a separate issue and get started on this!

**Comment by jorisvandenbossche** [comment]
Yes, I think that is a good idea as well. However, as a first step I would start with using more fixtures in each of the files (that can go together with parametrizing tests), and then later we can move fixtures we use in multiple places to a conftest.py file.

**Comment by cr458** [comment]
Ok, will get started on the other tests.
